src/audio/dai.c

```diff
@@ -108,16 +108,17 @@ static void dai_dma_cb(void *data, uint32_t type, struct dma_cb_data *next)
 		local_buffer = list_first_item(&dev->bsource_list,
 					       struct comp_buffer, sink_list);
 
-		dma_buffer_copy_to(local_buffer, dd->dma_buffer, dd->process,
-				   bytes);
+		dma_buffer_copy_to(local_buffer, bytes, dd->dma_buffer, bytes,
+				   dd->process, bytes / comp_sample_bytes(dev));
```

Comment:
division can be expensive, can we pre-calculate this value and just pass samples directly ?

Reply:
@lgirdwood Bytes are different every copy. We need to divide somewhere, so it won't matter whether I move it to processing function or leave it here. It's nothing new, since we divide in almost every processing component.